src/popup/components/QrCode.vue

```diff
@@ -0,0 +1,81 @@
+<template>
+  <div id="qr-canvas" />
+</template>
+<script>
+import QRCodeStyling from '@solana/qr-code-styling';
+
+import SHLogo from '../../icons/logo-smal-blue.png';
+
+export default {
+  props: {
+    value: { type: String, required: true },
+    size: { type: Number, required: true },
+  },
+  mounted() {
+    this.qrCode = new QRCodeStyling({
+      data: this.value,
+      width: this.size,
+      height: this.size,
+      margin: 0,
+      padding: 0,
+      qrOptions: {
+        typeNumber: '0',
+        mode: 'Byte',
+        errorCorrectionLevel: 'Q',
+        margin: 0,
+      },
+      imageOptions: {
+        hideBackgroundDots: true,
+        imageSize: 0.4,
+        margin: 0,
+      },
+      dotsOptions: { type: 'extra-rounded', color: '#000000' },
+      backgroundOptions: {
+        color: '#ffffff',
+        gradient: null,
+        margin: 0,
+        padding: 0,
+      },
+      image: SHLogo,
+      dotsOptionsHelper: {
+        colorType: { single: true, gradient: false },
+      },
+      cornersSquareOptions: { type: 'extra-rounded', color: '#000000' },
+      cornersSquareOptionsHelper: {
+        colorType: { single: true, gradient: false },
+        gradient: {
+          linear: true,
+          radial: false,
+          color1: '#000000',
+          color2: '#000000',
+          rotation: '0',
+        },
+      },
+      cornersDotOptions: { type: 'dot', color: '#000000' },
+      cornersDotOptionsHelper: {
+        colorType: { single: true, gradient: false },
+        gradient: {
+          linear: true,
+          radial: false,
+          color1: '#000000',
+          color2: '#000000',
+          rotation: '0',
+        },
+      },
+      backgroundOptionsHelper: {
+        colorType: { single: true, gradient: false },
+        gradient: {
+          linear: true,
+          radial: false,
+          color1: '#ffffff',
+          color2: '#ffffff',
+          rotation: '0',
+        },
+      },
+    });
+    this.qrCode.append(document.getElementById('qr-canvas'));
```

Comment:
Please avoid matching elements with the use of vanilla JS. In the scope of Vue components it's better to use references.